### ISSUE TITLE
SLEIGH builtin for swap

### ISSUE THREAD

**Issue by mumbel** [NationalSecurityAgency/ghidra#5959]
**Is your feature request related to a problem? Please describe.**
Endianess or other programming related issues that require bytes swaps produce hard to read code and worse ghidra doesn't pick up type information all of the time for the swapped output.

Almost all ISA would have this as an instruction, ghidra has varying implementations.

**Describe the solution you'd like**
SLEIGH built in pcode for swap.  Most likely this would be swap for 16, 32, and 64 (would this be 3 pcodeop or 1 where it picks up size?)

Ideally would see in the decompiler and implementing SLEIGH could be this string or just `swap<N >` maybe.

```
out = __builtin_bswap16(in);
out = __builtin_bswap32(in);
out = __builtin_bswap64(in);
```

**Describe alternatives you've considered**

**Additional context**
There have been comments in other issues and pull requests about coming features to easily extend pcodeops, don't know if this would be more common so makes sense to be in SLEIGH or if this sort of feature is easily solved with the approach being discussed elsewhere.

**Comment by LukeSerne** [comment]
Related: #306 (this seems to actually be a duplicate of #306, but this issue seems somewhat further in the issue pipeline, so it'd probably be better to keep this open)

I completely agree with this. The various multi-statement implementations in current languages are rarely optimised away by the decompiler, and the resulting pseudo-C statements are often interleaved with other, unrelated operations, making the decompiler output hard to follow.

I think a single operation `BYTESWAP(x, n)` would be better than having 3 byte-swapping opcodes . Keeping the number of pcode ops small is helpful to avoid code duplication in various places in the decompiler. Here, `x` is the varnode to swap and `n` is the number of bits of the sections that are byteswapped. So `__builtin_bswap16(in)` would translate to `BYTESWAP(in, 16)`.
